### DISCUSSION TITLE
Manually add function to the profile for `wasm-split`

### DISCUSSION THREAD

**Discussion by mmomtchev** [WebAssembly/binaryen#7030] [General]
Currently there does not seem to be any way to manually add select functions to the profile used for `wasm-split`?

In my case, I have environment finalizers that run when the environment is destroyed and which cannot be called before `write_profile`.

One solution would be to spawn the profiled code in a `worker_thread` - however my module does not use `SharedArrayBuffer` and I want to keep it this way.

Another solution would be to use `profile-print` and to generate a manifest to feed into `multi-split` - however `multi-split` seems to be slightly incompatible with `split` and the `emscripten` loader does not support it.

Has anyone done this before with the existing tools?

**Comment by mmomtchev** [comment]
I am wiling to contribute a PR for this problem, how about supporting `keep-func` with `profile`?

**Reply by mmomtchev** [reply]
Ok, I found a way to do this, use `print-profile`, then extract the list of the functions from the output, add the missing functions, and pass this new file with `--keep-funcs @...`

**Reply by tlively** [reply]
Nice workaround! If you still want do make a PR so you don't need that workaround, supporting `--keep-funcs` with `--profile` makes sense to me.
